blob/s3blob/s3blob.go

```diff
@@ -628,11 +663,33 @@ func unescapeKey(key string) string {
 // NewTypedWriter implements driver.NewTypedWriter.
 func (b *bucket) NewTypedWriter(ctx context.Context, key string, contentType string, opts *driver.WriterOptions) (driver.Writer, error) {
 	key = escapeKey(key)
-	uploader := s3manager.NewUploaderWithClient(b.client, func(u *s3manager.Uploader) {
-		if opts.BufferSize != 0 {
-			u.PartSize = int64(opts.BufferSize)
-		}
-	})
+
+	// Check if a cached uploader already exists.
+	b.RLock()
+	cachedUploader := b.cachedUploader
+	b.RUnlock()
+
+	var (
+		uploader    *s3manager.Uploader
+		uploaderKey = newUploaderKey(opts.BufferSize)
+	)
+	if cachedUploader != nil && uploaderKey == cachedUploader.key {
+		// If a cached uploader already exists *and* it has the same key/configuration
```

Comment:
So if some writes use one BufferSize, and other writes use another BufferSize, we'll never get to reuse the uploader? Maybe have a map instead of a single cached one?

Reply:
Please add a test for this new function as well.